src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/HardwareIntrinsicHelpers.Aot.cs

```diff
@@ -49,12 +49,28 @@ public static MethodIL EmitIsSupportedIL(MethodDesc method, FieldDesc isSupporte
             var emit = new ILEmitter();
             ILCodeStream codeStream = emit.NewCodeStream();
 
-            codeStream.Emit(ILOpcode.ldsfld, emit.NewToken(isSupportedField));
-            codeStream.EmitLdc(flag);
-            codeStream.Emit(ILOpcode.and);
-            codeStream.EmitLdc(0);
-            codeStream.Emit(ILOpcode.cgt_un);
-            codeStream.Emit(ILOpcode.ret);
+            if(!uint.IsPow2((uint)flag))
+            {
+                // These are the ISAs managed by multiple-bit flags.
+                // we need to emit different IL to handle the checks.
+                // (isSupportedField & flag) == flag
+                codeStream.Emit(ILOpcode.ldsfld, emit.NewToken(isSupportedField));
+                codeStream.EmitLdc(flag);
+                codeStream.Emit(ILOpcode.and);
+                codeStream.EmitLdc(flag);
+                codeStream.Emit(ILOpcode.beq);
```

Comment:
```suggestion
                codeStream.Emit(ILOpcode.ceq);
```
`beq` is "branch on equal". This is needs to be compare instead.

Reply:
Corrected, thanks for pointing out!